### ISSUE TITLE
Redirect reddit.com/ -> old.reddit.com/

### ISSUE THREAD

**Issue by anthonypkeane** [brave/brave-browser#35899]
## Description 
Some info
https://twitter.com/outlineaddict/status/1755629459791593980?s=12&t=ksQozRcixOu7hcHTVdxisQ
https://bravesoftware.slack.com/archives/C0816QH3R/p1707410212267579

Like https://github.com/brave/brave-ios/issues/5657 please create an option to redirect reddit.com/-> old.reddit.com/

@anthonypkeane or @timchilds to provide info on placement and Settings text.

**Comment by deeppandya** [comment]
> @deeppandya this would be a new screen under Settings -> General -> Website Redirects.
> 
> Two toggles:
> 
> **reddit → old.reddit.com**
> Automatically redirects Reddit links to the older interface. To force the new Reddit interface, use new.reddit.com instead.
> 
> **npr.org → text.npr.org**
> Automatically redirects NPR links to their text-based version.
> 
> Both disabled by default.
> 
> @bradleyrichter we'll need a "Website Redirects" icon. Perhaps we can repurpose the iOS icon?
> https://www.figma.com/file/SLFLpMd7F7Tu58qGBdcUQG?node-id=2720:20828&mode=design#708431015

Yeah we will be using the same icon. It would work fine for android.

**Comment by TechFanTheo** [comment]
Just for context, this is important because **Reddit** now **blocks** access to the site for **VPN** users under a bogus "network policy" and "user agent" message. Only way is to either login to reddit, or use old.reddit.com. 
But www.old.reddit.com will generate a safety error, so one has to delete the www and put old.reddit.com

**Comment by TechFanTheo** [comment]
> To force the new Reddit interface, use new.reddit.com instead.

This might not work, as new.reddit.com redirects to reddit.com, and therefore will redirect to old.reddit.com

Furthermore, this is unnecessary, just as when the AMP-redirect option is activated, we don't need to provide an exception, as users likely want the change to be permanent.

**Comment by bsclifton** [comment]
cc: @timchilds for prioritization

**Comment by mlntr** [comment]
> cc: [@timchilds](https://github.com/timchilds) for prioritization

Has @timchilds done the prioritisation yet?

**Comment by timchilds** [comment]
@mlntr this did not make the priority list for Q3. It will be considered again for Q4.

**Comment by kolon0** [comment]
> Just for context, this is important because **Reddit** now **blocks** access to the site for **VPN** users under a bogus "network policy" and "user agent" message. Only way is to either login to reddit, or use old.reddit.com. 
> But www.old.reddit.com will generate a safety error, so one has to delete the www and put old.reddit.com

However, old.reddit.com also blocks me when I use a VPN, and I'm on Android. I thought I could write a custom filter to redirect www.reddit.com to a Redlib instance, but I couldn't find such syntax support in Brave. So I have to rewrite the domain manually every time I want to access Reddit from Google.
